compiler/rustc_interface/src/queries.rs

```diff
@@ -148,12 +148,8 @@ impl<'tcx> Queries<'tcx> {
             );
             let dep_graph = setup_dep_graph(sess, crate_name, stable_crate_id)?;
 
-            let lint_store = Lrc::new(passes::create_lint_store(
-                sess,
-                &*self.codegen_backend().metadata_loader(),
-                self.compiler.register_lints.as_deref(),
-                &pre_configured_attrs,
-            ));
+            let lint_store =
+                Lrc::new(passes::create_lint_store(sess, self.compiler.register_lints.as_deref()));
```

Comment:
Doesn't need to be done right now, but maybe this can be moved into `build_session` now? I wouldn't be surprised if there are a couple of places where we are temporarily stashing lints just to wait on creating the lint store too. Those could be cleaned up too.

Reply:
Looking at this now... can you expand a little? I don't entirely understand the end point you are after. Would `GlobalCtxt` still hold the lint store?

Reply:
I have an in-progress change that moves the `lint_store` from `GlobalCtxt` to `Session`. Is that what you had in mind?

Reply:
> Is that what you had in mind?

Yes. There is a place in the driver where a lint store is created before TyCtxt. With this change it would be possible to use the lint store stored in Session instead. Do you think the end result will be nicer than the current state?

Reply:
I did this in #117649.